test_jpeg/test/test_rlecore.py

```diff
@@ -0,0 +1,130 @@
+from myhdl.conversion import *
+from myhdl import *
+from jpegenc.subblocks.RLE.RLECore.rlecore import DataStream, rle
+from jpegenc.subblocks.RLE.RLECore.rlecore import RLESymbols, RLEConfig
+from testcases import *
+
+red = 1
+green = 2
+blue = 3
+
+WIDTH_RAM_ADDRESS = 6
+WIDTH_RAM_DATA = 12
+
+
+def start_of_block(clock, start):
+    start.next = True
+    yield clock.posedge
+    start.next = False
+    yield clock.posedge
+
+
+def reset_on_start(clock, reset):
+    reset.next = True
+    yield delay(20)
+    yield clock.posedge
+    reset.next = False
+
+
+def block_process(clock, block, datastream, rlesymbols, rleconfig, color):
+    # select color component
+    rleconfig.color_component.next = color
+    yield start_of_block(clock, datastream.start)
+
+    datastream.input_val.next = block[rleconfig.read_addr]
+    yield clock.posedge
+    while rleconfig.read_addr != 63:
+        datastream.input_val.next = block[rleconfig.read_addr]
+        yield clock.posedge
+        if rlesymbols.dovalid:
+            print("amplitude = %d runlength = %d size = %d" % (
+                rlesymbols.amplitude, rlesymbols.runlength, rlesymbols.size))
+
+    datastream.input_val.next = block[rleconfig.read_addr]
+    
+    yield clock.posedge
+
+    if rlesymbols.dovalid:
+        print("amplitude = %d runlength = %d size = %d" % (
+            rlesymbols.amplitude, rlesymbols.runlength, rlesymbols.size))
+
+    yield clock.posedge
+    
+    if rlesymbols.dovalid:
+        print("amplitude = %d runlength = %d size = %d" % (
+            rlesymbols.amplitude, rlesymbols.runlength, rlesymbols.size))
+    
+    yield clock.posedge
+    
+    if rlesymbols.dovalid:
+        print("amplitude = %d runlength = %d size = %d" % (
+            rlesymbols.amplitude, rlesymbols.runlength, rlesymbols.size))
+    
+    yield clock.posedge
+    
+    if rlesymbols.dovalid:
+        print("amplitude = %d runlength = %d size = %d" % (
+            rlesymbols.amplitude, rlesymbols.runlength, rlesymbols.size))
+
+    yield clock.posedge
+    if rlesymbols.dovalid:
+        print("amplitude = %d runlength = %d size = %d" % (
+                rlesymbols.amplitude, rlesymbols.runlength, rlesymbols.size))
+
+
+@block
+def test():
+
+    clock = Signal(bool(0))
+    reset = ResetSignal(0, active=1, async=True)
+
+    datastream = DataStream()
+    rlesymbols = RLESymbols()
+    rleconfig = RLEConfig()
+
+
+    jpgv1 = rle(
+        reset, clock,
+        datastream, rlesymbols, rleconfig
+        )
+
+    @instance
+    def tbclock():
+        clock.next = 0
+        while True:
+            yield delay(10)
+            clock.next = not clock
+
+    @instance
+    def tbsim():
+        """ reset signal """
+        yield reset_on_start(clock, reset)
+
+        yield block_process(
+            clock, blue_pixels_1,
+            datastream,
+            rlesymbols,
+            rleconfig,  red
+            )
+
+        print ("=====")
+
+
```

Comment:
To check `Run Length Encoder` we need inputs like more than `15 continous zero's` more than 32 and all ... So, I made `testcases` file to add test cases for which we want to check .... Can you suggest me some other way to do it ?